x-pack/legacy/plugins/apm/public/components/app/Settings/AgentConfigurations/AddEditFlyout/ServiceSection.tsx

```diff
@@ -0,0 +1,152 @@
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { SelectWithPlaceholder } from '../../../../shared/SelectWithPlaceholder';
+import { useFetcher } from '../../../../../hooks/useFetcher';
+import { callApmApi } from '../../../../../services/rest/callApmApi';
+import { getOptionLabel, getOptionValue } from '../constants';
+
+const SELECT_PLACEHOLDER_LABEL = `- ${i18n.translate(
+  'xpack.apm.settings.agentConf.flyOut.serviceSection.selectPlaceholder',
+  { defaultMessage: 'Select' }
+)} -`;
+
+interface Props {
+  isReadOnly: boolean;
+  serviceName?: string;
+  setServiceName: (env: string) => void;
+  environment?: string;
+  setEnvironment: (env: string) => void;
+}
+
+export function ServiceSection({
+  isReadOnly,
+  serviceName,
+  setServiceName,
+  environment,
+  setEnvironment
+}: Props) {
+  const { data: serviceNames = [], status: serviceNamesStatus } = useFetcher(
+    () => {
+      if (!isReadOnly) {
+        return callApmApi({
+          pathname: '/api/apm/settings/agent-configuration/services',
+          forceCache: true
+        });
+      }
+    },
+    [isReadOnly],
+    { preservePreviousData: false }
+  );
+  const { data: environments = [], status: environmentStatus } = useFetcher(
+    () => {
+      if (!isReadOnly && serviceName) {
+        return callApmApi({
+          pathname: '/api/apm/settings/agent-configuration/environments',
+          params: { query: { serviceName: getOptionValue(serviceName) } }
+        });
+      }
+    },
+    [isReadOnly, serviceName],
+    { preservePreviousData: false }
+  );
+
+  const serviceNameOptions = serviceNames.map(name => ({
+    text: getOptionLabel(name),
+    value: name
+  }));
+  const environmentOptions = environments.map(
+    ({ name, alreadyConfigured }) => ({
+      disabled: alreadyConfigured,
+      text: `${getOptionLabel(name)} ${
+        alreadyConfigured ? '(already configured)' : ''
+      }`,
+      value: name
+    })
+  );
+
+  return (
+    <>
+      <EuiTitle size="xs">
+        <h3>
+          {i18n.translate(
+            'xpack.apm.settings.agentConf.flyOut.serviceSection.title',
+            { defaultMessage: 'Service' }
+          )}
+        </h3>
+      </EuiTitle>
+
+      <EuiSpacer size="m" />
+
+      <EuiFormRow
+        label={i18n.translate(
+          'xpack.apm.settings.agentConf.flyOut.serviceSection.serviceNameSelectLabel',
+          { defaultMessage: 'Name' }
+        )}
+        helpText={
+          !isReadOnly &&
+          i18n.translate(
+            'xpack.apm.settings.agentConf.flyOut.serviceSection.serviceNameSelectHelpText',
+            { defaultMessage: 'Choose the service you want to configure.' }
+          )
+        }
+      >
+        {isReadOnly ? (
+          <EuiText>{getOptionLabel(serviceName)}</EuiText>
+        ) : (
+          <SelectWithPlaceholder
+            placeholder={SELECT_PLACEHOLDER_LABEL}
+            isLoading={serviceNamesStatus === 'loading'}
```

Comment:
should we use constants here?

Reply:
I thought it was important to use constants but since `serviceNamesStatus` is typed as `FETCH_STATUS` which is `'loading' | 'success' | 'failure' | 'pending'` typescript will complain if the string doesn't match. 
If we use constants elsewhere I'll change it for consistency.